src/cbc_sdk/platform/observations.py

```diff
@@ -0,0 +1,548 @@
+            self._submit()
+
+        while self._still_querying():
+            time.sleep(0.5)
+
+        if self._timed_out:
+            raise TimeoutError(
+                message="user-specified timeout exceeded while waiting for results"
+            )
+
+        log.debug("Pulling results, timed_out={}".format(self._timed_out))
+
+        current = start
+        rows_fetched = 0
+        still_fetching = True
+        query_parameters = {}
+        result_url_template = (
+            "/api/investigate/v2/orgs/{}/observations/search_jobs/{}/results".format(
+                self._cb.credentials.org_key, self._query_token
+            )
+        )
+
+        while still_fetching:
+            result_url = "{}?start={}&rows={}".format(
+                result_url_template, current, self._batch_size
+            )
+            result = self._cb.get_object(result_url, query_parameters=query_parameters)
+            results = result.get("results", [])
+
+            self._total_results = result.get("num_available", 0)
+            self._count_valid = True
+
+            for item in results:
+                yield item
+                current += 1
+                rows_fetched += 1
+
+                if rows and rows_fetched >= rows:
+                    still_fetching = False
+                    break
+
+            if current >= self._total_results:
+                still_fetching = False
+
+            log.debug("current: {}, total_results: {}".format(current, self._total_results))
+
+    def get_group_results(
+        self,
+        fields,
+        max_events_per_group=None,
+        rows=500,
+        start=None,
+        range_duration=None,
+        range_field=None,
+        range_method=None
+    ):
+        """
+        Get group results grouped by provided fields.
+
+        Args:
+            fields (str / list): field or fields by which to perform the grouping
+            max_events_per_group (int):Maximum number of events in a group, if not provided, all events will be returned
+            rows (int): Number of rows to request, can be paginated
+            start (int): First row to use for pagination
+            ranges (dict): dict with information about duration, field, method
+
+        Returns:
+            dict: grouped results
+        """
+        if not isinstance(fields, list) and not isinstance(fields, str):
+            raise ApiError("Fields should be either a single field or list of fields")
+
+        if isinstance(fields, str):
+            fields = [fields]
+
+        if not all((gf in ObservationQuery.VALID_GROUP_FIELDS) for gf in fields):
+            raise ApiError("One or more invalid aggregation fields")
+
+        if not self._query_token:
+            self._submit()
+
+        result_url = "/api/investigate/v2/orgs/{}/observations/search_jobs/{}/group_results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+
+        data = self._build_grouped_body(
```

Comment:
Is this needed as a separate function? Do we plan on re-using this somewhere?

Reply:
Not really, moving the code into get_group_results.